app/services/metrics/weekly/all_metrics.rb

```diff
@@ -0,0 +1,50 @@
+module Metrics
+  module Weekly
+    class AllMetrics < BaseService
+      def call
+        process
+      end
+
+      private
+
+      def process
+        Metric.today_daily_metrics.find_each do |daily_metric|
```

Comment:
This method is `O(n2)` with the `Metrics` table, which could grow a lot. I think we can move forward with this solution though, because I can't think of a better one, but we're going to have to keep an eye on the performance of this job.

Reply:
roger that